pkg-manifest/exportable-manifest/src/index.ts

```diff
@@ -16,29 +17,42 @@ const PREPUBLISH_SCRIPTS = [
 ]
 
 export interface MakePublishManifestOptions {
+  catalogResolver: CatalogResolver
   modulesDir?: string
   readmeFile?: string
 }
 
 export async function createExportableManifest (
   dir: string,
   originalManifest: ProjectManifest,
-  opts?: MakePublishManifestOptions
+  opts: MakePublishManifestOptions
 ): Promise<ProjectManifest> {
   const publishManifest: ProjectManifest = omit(['pnpm', 'scripts', 'packageManager'], originalManifest)
   if (originalManifest.scripts != null) {
     publishManifest.scripts = omit(PREPUBLISH_SCRIPTS, originalManifest.scripts)
   }
+
+  const replaceCatalogProtocol = (depName: string, depSpec: string) =>
+    resolveCatalogProtocol(opts.catalogResolver, depName, depSpec)
```

Comment:
Why not just using bind?

```suggestion
  const replaceCatalogProtocol = resolveCatalogProtocol.bind(null, opts.catalogResolver)
```

Reply:
The only reason is because I'm not used to using `.bind`. 😅

To be honest, I personally prefer the plain arrow functions here since you don't have to re-arrange parameters, but I can adjust when writing code in pnpm.